src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityTests.java

```diff
@@ -70,6 +84,76 @@ public static void initIndexes() throws Exception {
         Collections.sort(indexes);
     }
     
+    @AfterClass
+    public static void tearDownStatics() {
+        indexes = null;
+        indicesDir = null;
+    }
+    
+    @Override
+    public Settings nodeSettings(int ord) {
+        return ImmutableSettings.builder()
+            .put(Node.HTTP_ENABLED, true) // for _upgrade
+            .put(MergePolicyModule.MERGE_POLICY_TYPE_KEY, NoMergePolicyProvider.class) // disable merging so no segments will be upgraded
+            .build();
+    }
+    
+    void setupDataLoadingNodes() throws Exception {
+        Path dataDir = newTempDirPath(LifecycleScope.SUITE);
+        indicesDir = dataDir.resolve(internalCluster().getClusterName()).resolve("nodes/0/indices/");
```

Comment:
@rjernst this is what I meant : https://github.com/bleskes/elasticsearch/commit/1e38fac385757fac62f0ab0532d557d804caddab 

Imho it's cleaner as it full relies on the test cluster to setup things. If we randomize to mutilpe data paths (which we should, it will randomly select one for the injection. Note that the commit also have formatting fixed, so it will be bigger).